lib/package-manager/get-executable.js

```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+const path = require('path');
+const which = require('which');
+const npmWhich = require('npm-which')(__dirname);
+
+const windows = (process.platform === 'win32');
+
+module.exports = function getExecutable(binaryName, next) {
+  if (binaryName === 'yarn') {
+    // Use `npm-which` for yarn to get the locally version
+    npmWhich(binaryName, next);
```

Comment:
is there a reason we can't use npmWhich for both `npm` and `yarn`?. If that's the case and it is pass through could we not remove this script altogether?

Reply:
>is there a reason we can't use npmWhich for both `npm` and `yarn`?

`npm-which` is just for binaries installed in `node_modules`. Not sure if it finds globally installed stuff?

Could potentially add `npm` as a dep as well, then it won't rely on any globals beyond node. Maybe not worth it?

> If that's the case and it is pass through could we not remove this script altogether?

IDK about the windows thing below. Would that just work :tm:? I don't have access to a windows machine to test

Reply:
Let's keep it just how it is for now, this makes sense. I was more confused by the usage above